main.py

```diff
@@ -0,0 +1,176 @@
+    for i in range(0, len(new_values1)):
+        new_values1[i] = int(new_values1[i])
+
+    new_values2 = [list for key, list in dict2.items()]
+    for i in range(0, len(new_values2)):
+        new_values2[i] = int(new_values2[i])
+
+    print("Highest Average: " + str(ceil(mean(new_values))) + "C")
+    print("Lowest Average: " + str(ceil(mean(new_values1))) + "C")
+    print("Average Mean Humidity: " + str(ceil(mean(new_values2))) + "%")
+
+
+def yearly_values(dict1, dict2, dict3):
+    max_temprature = max(dict1.values())
+    key_list = list(dict1.keys())
+    val_list = list(dict1.values())
+
+    # print key with val temp
+    position = val_list.index(max_temprature)
+    print("Highest: " + max_temprature + "C on " + key_list[position])
+
+    min_temprature = max(dict2.values())
+    key_list = list(dict2.keys())
+    val_list = list(dict2.values())
+
+    # print key with val temp
+    position = val_list.index(min_temprature)
+    print("Lowest: " + min_temprature + "C on " + key_list[position])
+
+    max_humidity = max(dict1.values())
+    key_list = list(dict1.keys())
+    val_list = list(dict1.values())
+
+    # print key with val temp
+    position = val_list.index(max_humidity)
+    print("Humidity: " + max_humidity + "% on " + key_list[position])
+
+
+def file_processing(i, files_list):
+    index = i
+    files_list = files_list
+    files_path = "/home/hamza/Downloads/weatherfiles/weatherfiles/"
+    filepaths = [os.path.join(files_path, name) for name in os.listdir(files_path)]
+    print(files_path)
+    for i in files_list:
+        if i in files_list:
+            with open(i) as f:
+                csv_file = reader(f)
+                next(csv_file)
+                dict_rows_key = {rows[0]: rows[9] for rows in csv_file}
+    return dict_rows_key
+
+
+def file_parsing(year, month_name, option):
+    year = year
+    month = month_name
+    months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    files_list = [None] * 12
+    j = 0
+    if option == "-e":
+        list_length = len(files_list)
+
+        for i in range(list_length):
+            files_list[i] = ("Murree_weather_" + year + "_" + months[j] + ".txt")
+            j = j + 1
+
+    else:
+        files_list[0] = "Murree_weather_" + year + "_" + months[j] + ".txt"
+    dict_max_temprature = file_processing(1, files_list)
+    dict_min_temprature = file_processing(3, files_list)
+    dict_humidity = file_processing(9, files_list)
+    if year != month_name:
+        monthly_average_values(dict_max_temprature, dict_min_temprature,
+                               dict_humidity)
+
+    elif option == "-a":
+        yearly_values(dict_max_temprature, dict_min_temprature, dict_humidity)
+    elif option == "-c":
+        get_first_second_temprature(dict_max_temprature, dict_min_temprature, month, year)
+
+
+if __name__ != '__main__':
+    pass
+else:
+    main(sys.argv[1])
```

Comment:
Read the tasks documentation properly, what we are reading as command line arguments and manage them well instead of getting one from it.